### ISSUE TITLE
Exported enums do not allow default value

### ISSUE THREAD

**Issue by rosshadden** [godotengine/godot#15585]
**Godot version:**

`master` branch, commit 554e95747, but really all versions

**OS/device including version:**

Arch Linux

**Issue description:**

As the title says, when exporting a variable with an enum type, it does not allow the value to be set.

**Steps to reproduce:**

Given this code:

```gdscript
enum Foobar { FOO, BAR }
export(Foobar) var foobar = Foobar.BAR

func _ready():
  print(foobar)
```

`foobar` is `null`.

**Comment by vnen** [comment]
CC @bojidar-bg

**Comment by rosshadden** [comment]
I think the problem is within [this block](https://github.com/godotengine/godot/blob/e4213e66b2dd8f5a87d8cf5015ac83ba3143279d/modules/gdscript/gdscript_parser.cpp#L3844-L3881), but I haven't yet figured out what to change or add. If someone were to provide a bit of insight I can probably fix it myself.

**Comment by bojidar-bg** [comment]
No idea what might have caused it, it should work fine...
This block should just make a `PropertyInfo(Variant::INT, "..", HINT_ENUM, "text:0,text:1..")`

**Comment by vnen** [comment]
The problem lies somewhere else, because it shows the value in editor, it's just Null when running the code.

**Comment by KennethWilke** [comment]
👍 ran into this as well

```
enum Direction {UP, DOWN, LEFT, RIGHT}
export(Direction) var facing = DOWN
```

Assigning my integer value as the default for now so that I get the behavior I expected in the editor
```
enum Direction {UP, DOWN, LEFT, RIGHT}
export(Direction) var facing = 1
```

**Comment by bfloch** [comment]
Just hit the same. Thanks for the workaround @KennethWilke

**Comment by vonflyhighace2** [comment]
Yup this is a sigh one that I had the fortunate headache of running into.

**Comment by bojidar-bg** [comment]
Potentially fixed by #19635, but needs #19534 to be fixed before it can be tested :(.

**Comment by rminderhoud** [comment]
@bojidar-bg I believe #19534 was sufficient and this can be closed. It is working for me on the latest master without #19635

**Comment by bojidar-bg** [comment]
Ok, closing as fixed by #19635. If it is still an issue, we might reopen it later on.
